File: app/src/test/java/com/chetdeva/madrasi/root/onboarding/OnboardingRouterTest.java
package com.chetdeva.madrasi.root.onboarding;

import com.ubercab.test.UberTestBase;
import com.uber.rib.core.RouterHelper;

import org.junit.Before;
import org.junit.Test;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;

public class OnboardingRouterTest extends UberTestBase {

  @Mock private LandingScope scope;
  @Mock private OnboardingInteractor interactor;
  @Mock private OnboardingView view;

  private OnboardingRouter router;

  @Before
  public void setup() {
    MockitoAnnotations.initMocks(this);

    router = new OnboardingRouter(scope, view, interactor);
  }

  /**
   * TODO: Delete this example and add real tests.
   */
  @Test
  public void anExampleTest_withSomeConditions_shouldPass() {
    // Use RouterHelper to drive your router's lifecycle.
    RouterHelper.attach(router);
    RouterHelper.detach(router);

    throw new RuntimeException("Remove this test and add real tests.");
  }

}

File: app/src/androidTest/java/com/chetdeva/madrasi/root/onboarding/OnboardingRouterTest.kt
package com.chetdeva.madrasi.root.onboarding

import com.uber.rib.core.RibTestBasePlaceholder
import com.uber.rib.core.RouterHelper

import org.junit.Before
import org.junit.Test
import org.mockito.Mock
import org.mockito.MockitoAnnotations

class OnboardingRouterTest : RibTestBasePlaceholder() {

  @Mock internal lateinit var component: OnboardingBuilder.Component
  @Mock internal lateinit var interactor: OnboardingInteractor
  @Mock internal lateinit var view: OnboardingView

  private var router: OnboardingRouter? = null

  @Before
  fun setup() {
    MockitoAnnotations.initMocks(this)

    router = OnboardingRouter(view, interactor, component)
  }

  /**
   * TODO: Delete this example and add real tests.
   */
  @Test
  fun anExampleTest_withSomeConditions_shouldPass() {
    // Use RouterHelper to drive your router's lifecycle.
    RouterHelper.attach(router!!)
    RouterHelper.detach(router!!)

    throw RuntimeException("Remove this test and add real tests.")
  }

}

